content/en/docs/reference/access-authn-authz/authorization.md

```diff
@@ -4,247 +4,206 @@ reviewers:
-```
+<a id="warning-always-allow" />
 
-```shell
-kubectl auth can-i create deployments --namespace prod
-```
+{{< warning >}}
+Enabling the `AlwaysAllow` mode bypasses authorization; do not use this on a cluster where
+you do not trust **all** potential API clients, including the workloads that you run.
 
-The output is similar to this:
-
-```
-no
-```
-
-Administrators can combine this with [user impersonation](/docs/reference/access-authn-authz/authentication/#user-impersonation)
-to determine what action other users can perform.
-
-```bash
-kubectl auth can-i list secrets --namespace dev --as dave
-```
-
-The output is similar to this:
+You should not use the `AlwaysAllow` mode on a Kubernetes cluster where the API server
+is reachable from the public internet.
+{{< /warning >}}
 
-```
-no
-```
-
-Similarly, to check whether a ServiceAccount named `dev-sa` in Namespace `dev`
-can list Pods in the Namespace `target`:
-
-```bash
-kubectl auth can-i list pods \
-	--namespace target \
-	--as system:serviceaccount:dev:dev-sa
-```
-
-The output is similar to this:
-
-```
-yes
-```
-
-`SelfSubjectAccessReview` is part of the `authorization.k8s.io` API group, which
-exposes the API server authorization to external services. Other resources in
-this group include:
-
-* `SubjectAccessReview` - Access review for any user, not only the current one. Useful for delegating authorization decisions to the API server. For example, the kubelet and extension API servers use this to determine user access to their own APIs.
-* `LocalSubjectAccessReview` - Like `SubjectAccessReview` but restricted to a specific namespace.
-* `SelfSubjectRulesReview` - A review which returns the set of actions a user can perform within a namespace. Useful for users to quickly summarize their own access, or for UIs to hide/show actions.
-
-These APIs can be queried by creating normal Kubernetes resources, where the response "status"
-field of the returned object is the result of the query.
+You can configure the Kubernetes API server's authorizer chain using
+either a command line setting or, as an alpha feature, using a
+configuration file.
 
-```bash
-kubectl create -f - -o yaml << EOF
-apiVersion: authorization.k8s.io/v1
-kind: SelfSubjectAccessReview
-spec:
-  resourceAttributes:
-    group: apps
-    resource: deployments
-    verb: create
-    namespace: dev
-EOF
-```
+### Command line authorization mode configuration {#using-flags-for-your-authorization-module}
 
-The generated `SelfSubjectAccessReview` is:
-```yaml
-apiVersion: authorization.k8s.io/v1
-kind: SelfSubjectAccessReview
-metadata:
-  creationTimestamp: null
-spec:
-  resourceAttributes:
-    group: apps
-    resource: deployments
-    namespace: dev
-    verb: create
-status:
-  allowed: true
-  denied: false
-```
+You must set a command line argument to the API server, to indicate which authorizationmode
+the API server should use.
 
-## Using Flags for Your Authorization Module
+You can use the following modes:
```

Comment:
“flags” is a confusing term for people who don't code in Go. To the wider world of IT, flags usually pecify _boolean_ values. Golang has a different convention, but we can't assume the reader is a Go coder.

Reply:
Actually, the examples below are command line flags or options. I agree to @ritazh. It is not a Go thing. It is a k8s convention.

Reply:
Because "flags” is a confusing term for people who don't code in Go, I've been working to amend the docs so that they are more widely understood, matching common usage in the world of IT. See https://superuser.com/a/1070071 for an example.

Kubernetes has copied Go's convention here, but I'm asserting it's an unhelpful convention. Early on, people who used Kubernetes were often Go coders, so contributors didn't notice.